pennylane_lightning_kokkos/src/util/LinearAlgebra.hpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2022 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @file LinearAlgebra.hpp
+ * Contains uncategorised utility functions.
+ */
```

Comment:
Could you please add a description for this file?
I see that you essentially re-wrote some methods already present in Kokkos Kernels.
May I know why? I mean, why not build Kokkos Kernels and make use of their functionality?
Did you compare performance?

Reply:
Lee and I discussed about this before. We decided to use the our own kernel now.

Reply:
Performance and functionality (support) may be a problem in the future, but I guess this is acceptable for now.
Thank you for clarifying. :slightly_smiling_face:

Reply:
Agree!